### PR TITLE
Source Postgres : Fast query for estimate messages

### PR DESCRIPTION
**Pull request by akashkulk**
As part of https://github.com/airbytehq/airbyte/pull/20783, trace estimate messages were added to source-postgres. 

One of the follow-up items https://github.com/airbytehq/airbyte/issues/21499 identified in that PR was to use a fast query to calculate number of rows instead of `select count(*)` which can be slow. This was initially thought to not be a problem : see https://github.com/airbytehq/airbyte/pull/20783#issuecomment-1384698480, however we are seeing user reports of this query being slow : https://github.com/airbytehq/airbyte/pull/20783#issuecomment-1399031995. 

In this PR, we migrate to the fast query. Initially, this was held off because invalid results were returned for smaller tables. But thinking more on this, we can skip estimate traces altogether for small tables (the sync will probably complete quickly anyways). This can be determined if the fast table row count estimate is something invalid (e.g. a negative value)

The only open question here is whether we should disable estimate trace messages for non-CDC incremental mode as I can't think of another way to get around this other than to issue a `select count(*) where cursor_id > cursor_value` query, which would have the same issues as a `select count(*)`

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public class PostgresQueryUtils {
 
   public static final String TABLE_ESTIMATE_QUERY =
       """
-            SELECT (SELECT COUNT(*) FROM %s) AS %s,
+            SELECT (select reltuples::int8 as count from pg_class c JOIN pg_catalog.pg_namespace n ON n.oid=c.relnamespace where nspname='%s' AND relname='%s') AS %s,
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java:29]: Maybe both versions should be kept and done in a try/catch... I wonder if there are certain older versions of postgres that don't have a `pg_catalog` table?

**Review by evantahler — APPROVED** [review]
👍 from me, but a question below

**Review by evantahler — APPROVED** [review]

**Comment by akashkulk** [comment]
/publish connector=connectors/source-postgres-strict-encrypt run-tests=false

**Comment by akashkulk** [comment]
/publish connector=connectors/source-postgres run-tests=false

**Comment by akashkulk** [comment]
/publish connector=connectors/source-postgres-strict-encrypt run-tests=false

**Pull request merged**